src/metavision_wrapper.cpp

```diff
@@ -428,8 +438,38 @@ void MetavisionWrapper::setDecodingEvents(bool decodeEvents)
   }
 }
 
+void MetavisionWrapper::activateTrailFilter()
+{
+  Metavision::I_EventTrailFilterModule * i_trail_filter =
+    cam_.get_device().get_facility<Metavision::I_EventTrailFilterModule>();
+
+  Metavision::I_EventTrailFilterModule::Type filter_type;
+
+  if (trailFilter_.type == TrailFilter::Type::TRAIL) {
+    filter_type = Metavision::I_EventTrailFilterModule::Type::TRAIL;
+  } else if (trailFilter_.type == TrailFilter::Type::STC_CUT_TRAIL) {
+    filter_type = Metavision::I_EventTrailFilterModule::Type::STC_CUT_TRAIL;
+  } else if (trailFilter_.type == TrailFilter::Type::STC_KEEP_TRAIL) {
+    filter_type = Metavision::I_EventTrailFilterModule::Type::STC_KEEP_TRAIL;
+  }
```

Comment:
Use a const static std::map<std::string, Metavision::I_EventTrailFilterModule::Type> map that you initialize with {}. This will make the code more readable. Then it = map.find(type) and throw exception if it == map.end().
Note again here you are not handling the case where all if()'s evaluate to false (but then you are using an enum, one can argue it's not necessary). Either way, this code should go in favor of a map.